bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/Converter.java

```diff
@@ -0,0 +1,71 @@
+package org.dataflowanalysis.analysis.converter;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceImpl;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.eclipse.emf.ecore.xmi.XMLResource;
+import org.eclipse.emf.ecore.xmi.impl.XMLResourceFactoryImpl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public abstract class Converter {
+    protected ObjectMapper objectMapper;
+    protected File file;
+
+    protected final Logger logger = Logger.getLogger(Converter.class);
+
+    public Converter() {
+        objectMapper = new ObjectMapper();
+    }
+
+    public void store(DataFlowDiagramAndDictionary complete, String outputFile) {
+        String truncatedOutputFile;
+        if (outputFile.endsWith(".json")) {
+            truncatedOutputFile = outputFile.substring(0, outputFile.length() - 5);
+        } else {
+            truncatedOutputFile = outputFile;
+        }
+        ResourceSet resourceSet = new ResourceSetImpl();
+        Resource dfdResource = createAndAddResource(truncatedOutputFile + ".dataflowdiagram", new String[] {"dataflowdiagram"}, resourceSet);
+        Resource ddResource = createAndAddResource(truncatedOutputFile + ".datadictionary", new String[] {"datadictionary"}, resourceSet);
+
+        dfdResource.getContents().add(complete.dataFlowDiagram());
+        ddResource.getContents().add(complete.dataDictionary());
+
+        saveResource(dfdResource);
+        saveResource(ddResource);
+
+    }
+
+    private Resource createAndAddResource(String outputFile, String[] fileextensions, ResourceSet resourceSet) {
+        for (String fileextension : fileextensions) {
+            resourceSet.getResourceFactoryRegistry().getExtensionToFactoryMap().put(fileextension, new XMLResourceFactoryImpl());
+        }
+        URI uri = URI.createFileURI(outputFile);
+        Resource resource = resourceSet.createResource(uri);
+        ((ResourceImpl) resource).setIntrinsicIDToEObjectMap(new HashMap<>());
```

Comment:
see other outdated comment. Why are you doing this?

Reply:
What do you mean by `this`?
This function is from the sdq wiki

Reply:
I see the problem :D. Setting this map should not be necessary, probably outdated. A long time ago, there where performance issues when loading large models. I think this has not been an issue for a very long time.
This was one of my old comments regarding the intrinsicidtoeobjectmap:

> Why are you even doing this? I have never seen this done and it should not be necessary. Please remove or add a comment why this is needed.
(I have seen using the intrinsic id to eObject map to improve loading performance, however, you should not have to set this one by hand...)